woocommerce/woocommerce-android
# Feature/10023 blaze analytics

Closes: #10023


### Description

Adds tracking for Blaze iteration 2 and removes tracking from prior implementation. 

The tracking plan followed was this: 
![Screenshot 2023-10-25 at 16 50 15](https://github.com/woocommerce/woocommerce-android/assets/2663464/cad7e71a-b6ec-447a-a9be-b2df1aa6c37f)


### Testing instructions

Follow the described steps and verify in the logcat that the indicated events are tracked. 

**Scenario 1**: No existing Blaze campaigns 

1. Log into Wp.com site (or Jetpack connected) that has now prior Blaze campaigns. 
2. Publish at least on product
3. Open my store tab: `blaze_entry_point_displayed, Properties: {"source":"my_store_section"`
4. Tap on "Create campaign": `blaze_intro_displayed, Properties: {"source":"my_store_section"...`
5. Click on "Start Blaze campaign now":  `blaze_entry_point_tapped, Properties: {"source":"intro_view"...` and `blaze_flow_started, Properties: {"source":"intro_view"`
6. Cancel Blaze flow by navigating back: `blaze_flow_canceled, Properties: {"source":"intro_view"`
7. Go to product detail: `blaze_entry_point_displayed, Properties: {"source":"product_detail_promote_button"`
8. Tap on Blaze CTA: `blaze_intro_displayed, Properties: {"source":"product_detail_promote_button"` 
9. Click on Start Blaze campaign now": `blaze_entry_point_tapped, Properties: {"source":"intro_view"`
10. Go to More menu: `blaze_entry_point_displayed, Properties: {"source":"menu"`
11. Tap on Blaze cta: `blaze_intro_displayed, Properties: {"source":"menu"`
12. Click on Start Blaze campaign now": `blaze_entry_point_tapped, Properties: {"source":"intro_view"`
13. Create the Blaze campaign: `blaze_flow_completed, Properties: {"source":"intro_view"`

**Scenario 2:** Existing Blaze campaigns
1. Reuse the previous site or log into a site that has prior Blaze campaigns
2. From My Store tap on "Show all": `blaze_campaign_list_entry_point_selected, Properties: {"source":"my_store_section"`
3. Tap on campaign detail: `blaze_campaign_detail_selected, Properties: {"source":"campaign_list"`
4. Navigate back to campaign list and tap on FAB: `blaze_entry_point_tapped, Properties: {"source":"campaign_list"` and `blaze_flow_started, Properties: {"source":"campaign_list"`
5. Navigate back to My Store tab and tap on "Create campaign": `blaze_entry_point_tapped, Properties: {"source":"my_store_section",` and `blaze_flow_started, Properties: {"source":"my_store_section"`
6. Go to More menu and tab Blaze cta: `blaze_campaign_list_entry_point_selected, Properties: {"source":"menu"`
